File: bandlog-team/src/main/java/net/effize/bandlog/team/controller/TeamController.java
package net.effize.bandlog.team.controller;

import net.effize.bandlog.shared.auth.AuthUser;
import net.effize.bandlog.team.dto.request.CreateTeamRequest;
import net.effize.bandlog.team.dto.request.JoinTeamRequest;
import net.effize.bandlog.team.dto.request.RefreshTeamInviteCodeRequest;
import net.effize.bandlog.team.dto.response.CreateTeamResponse;
import net.effize.bandlog.team.dto.response.JoinTeamResponse;
import net.effize.bandlog.team.dto.response.RefreshTeamInviteCodeResponse;
import net.effize.bandlog.team.dto.response.TeamInfoResponse;
import net.effize.bandlog.team.dto.response.TeamsResponse;
import net.effize.bandlog.team.model.UserId;
import net.effize.bandlog.team.service.TeamCommandService;
import net.effize.bandlog.team.service.TeamQueryService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/teams")
public class TeamController {
    private final TeamQueryService teamQueryService;
    private final TeamCommandService teamCommandService;


    public TeamController(TeamQueryService teamQueryService, TeamCommandService teamCommandService) {
        this.teamQueryService = teamQueryService;
        this.teamCommandService = teamCommandService;
    }

    @PostMapping
    public CreateTeamResponse createTeam(AuthUser authUser, @RequestBody CreateTeamRequest request) {
        return teamCommandService.createTeam(new UserId(authUser.id()), request);
    }

    @PostMapping("/refresh-invite-code")
    public RefreshTeamInviteCodeResponse refreshTeamInviteCode(AuthUser authUser, @RequestBody RefreshTeamInviteCodeRequest request) {
        return teamCommandService.refreshTeamInviteCode(new UserId(authUser.id()), request);
    }

    @PostMapping("/join")
    public JoinTeamResponse joinTeam(AuthUser authUser, @RequestBody JoinTeamRequest request) {
        return teamCommandService.joinTeam(new UserId(authUser.id()), request);
    }

    @GetMapping("/{teamId}")
    public TeamInfoResponse getTeamInfoResponse(AuthUser authUser, @PathVariable Long teamId) {
        return teamQueryService.teamInfo(new UserId(authUser.id()), teamId);
    }

    @GetMapping("/me")
    public TeamsResponse getMyTeams(AuthUser authUser) {
        return teamQueryService.myTeams(new UserId(authUser.id()));
    }
}

File: src/main/kotlin/net/effize/bandlog/team/adapter/in/web/TeamController.kt
package net.effize.bandlog.team.adapter.`in`.web

import net.effize.bandlog.common.auth.AuthUser
import net.effize.bandlog.common.id.UserId
import net.effize.bandlog.team.adapter.`in`.web.request.CreateTeamRequest
import net.effize.bandlog.team.adapter.`in`.web.request.JoinTeamRequest
import net.effize.bandlog.team.adapter.`in`.web.request.RefreshTeamInviteCodeRequest
import net.effize.bandlog.team.adapter.`in`.web.response.CreateTeamResponse
import net.effize.bandlog.team.adapter.`in`.web.response.JoinTeamResponse
import net.effize.bandlog.team.adapter.`in`.web.response.RefreshTeamInviteCodeResponse
import net.effize.bandlog.team.adapter.`in`.web.response.TeamInfoResponse
import net.effize.bandlog.team.adapter.`in`.web.response.TeamsResponse
import net.effize.bandlog.team.application.TeamCommandService
import net.effize.bandlog.team.application.TeamQueryService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/teams")
class TeamController(
    private val teamQueryService: TeamQueryService,
    private val teamCommandService: TeamCommandService
) {

    @PostMapping
    fun createTeam(authUser: AuthUser, @RequestBody request: CreateTeamRequest): CreateTeamResponse {
        return teamCommandService.createTeam(UserId(authUser.id), request)
    }

    @PostMapping("/refresh-invite-code")
    fun refreshTeamInviteCode(authUser: AuthUser, @RequestBody request: RefreshTeamInviteCodeRequest): RefreshTeamInviteCodeResponse {
        return teamCommandService.refreshTeamInviteCode(UserId(authUser.id), request)
    }

    @PostMapping("/join")
    fun joinTeam(authUser: AuthUser, @RequestBody request: JoinTeamRequest): JoinTeamResponse {
        return teamCommandService.joinTeam(UserId(authUser.id), request)
    }

    @GetMapping("/{teamId}")
    fun getTeamInfoResponse(authUser: AuthUser, @PathVariable teamId: Long): TeamInfoResponse {
        return teamQueryService.teamInfo(UserId(authUser.id), teamId)
    }

    @GetMapping("/me")
    fun getMyTeams(authUser: AuthUser): TeamsResponse {
        return teamQueryService.myTeams(UserId(authUser.id))
    }
}
